### PR TITLE
Add social filter

### PR DESCRIPTION
**Pull request by TheoBrigitte**
This PR adds the capability to check if a token has at least one social referenced.

This is configurable via the `CHECK_IF_SOCIAL` environment variable.
And logic is bundled within the mutable filter because social link is found within the token's metadata.

### CODE REVIEW

**Comment by fdundjer** [comment]
@TheoBrigitte Looks good, I'll push some changes to the same branch. It's hard to follow the logic with all those ternary operators.

**Comment by TheoBrigitte** [comment]
> @TheoBrigitte @fdundjer Just tried this branch and this does not do much, Social passed the check and got rugged instantly.
> 
> For this token, https://solscan.io/token/GqJVhdQwMuZsvEYcap78AJ7gvM1dY4qsw9yS91aswoWH https://bafkreihhw45ogcbukdwjjm6xdkak72pvqbibm3o7hq4syxoeskkfpgxysy.ipfs.nftstorage.link/
> 
> Dev copy/paste metadata from another token with the same name that has socials : https://twitter.com/ton_tepe
> 
> Cannot rely on anything in the metadata...

Social filter does not guarantee there won't be a rug, no filter can guarantee that. This is just another filter to help in the hunt for the gem's, up to you to combine filter according to your strategy.
